llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp

```diff
@@ -3832,7 +3832,8 @@ static bool checkForMustTailInVarArgFn(bool IsVarArg, const BasicBlock &BB) {
 
 bool IRTranslator::runOnMachineFunction(MachineFunction &CurMF) {
   MF = &CurMF;
-  const Function &F = MF->getFunction();
+  Function &F = MF->getFunction();
+  F.renumberBlocks(); // renumber blocks for dense values
```

Comment:
Is this safe? I think machine passes will preserve all the IR analyses, including DT, which may still get used by machine passes (e.g. through AA).

Reply:
Well, I guess this will just get handled when actually enabling the use of IR BB numbers in DT.

Reply:
When we switch the DT to use block numbers, use of outdated numbers will give assertion failures. I don't want to add renumbering to X other passes unless I can't add the then semi-required DT.updateBlockNumbers() at the same time.

I could remove renumbering here for now, but not sure if that's a good idea.

Reply:
I removed renumbering from this patch after all. The correct place would be the eliminate-unreachable-blocks pass, which will get a renumbering later.